### PR TITLE
System-actions

### PR DESCRIPTION
**Pull request by Ichmed**

### CODE REVIEW

File: core/src/builtin/ctl.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+                buf.clear();
+            }
+        }
+    })
+}
+
+async fn create_pipe() -> Result<BufReader<File>> {
+    let dir = if cfg!(debug_assertions) {
+        "test"
```
Comment (isbm) [inline core/src/builtin/ctl.rs:80]: ? 😉

File: core/src/config/mod.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        self.after.push(name.to_owned());
+        self
+    }
+}
+
+pub fn read_config(builtin: Vec<TaskConfigYaml>) -> Vec<TaskConfig> {
+    let configs = if cfg!(debug_assertions) {
+        "test"
+    } else {
+        "/etc/alfad"
```
Comment (isbm) [inline core/src/config/mod.rs:71]: Probably this is a good idea to have these as constants somewhere, so `Path::new(somewhere.DIR_CFG)` for example.

Comment (isbm) [reply core/src/config/mod.rs:71]: @Ichmed BTW, what's wrong with:

```rust
let configs = Path::new(if cfg!(debug_assertions) { somewhere.DIR_TEST } else { somewhere.DIR_CFG });
```

...?

File: core/src/config/mod.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    };
+    let configs = Path::new(configs);
+
+    match read_binary(configs.join("alfad.bin").as_path()) {
```
Comment (isbm) [inline core/src/config/mod.rs:75]: `somewhere.CFG_BIN` ?

File: core/src/config/mod.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        Some(mut configs) => {
+            configs.extend(
+                builtin
+                    .into_iter()
+                    .map(TaskConfigYaml::into_config)
+                    .filter_map(drop_errors),
+            );
+            configs
+        }
+        None => read_yaml_configs(configs.join("alfad.d").as_path(), builtin),
```
Comment (isbm) [inline core/src/config/mod.rs:85]: `somewhere.DIR_CFG_D` ?

File: core/src/config/payload.rs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+
+impl<T: FromStr> FromStr for Payload<T> {
+    type Err = <T as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self::Normal(T::from_str(s)?))
+    }
+}
```
Comment (isbm) [inline core/src/config/payload.rs:74]: Newline missing. 😉

File: core/src/init.rs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                        // Ignore Err(_) since ECHILD is expected
+                        waitpid(Some(Pid::from_raw(proc.pid)), None).ok();
+                    }
+                    _ => {}
+                }
+            }
+        })
+        .detach();
+
+        env::set_var("SMOL_THREADS", "8");
```
Comment (isbm) [inline core/src/init.rs:45]: I would do that _only_ if `SMOL_THREADS` was not defined yet from the outside.

Comment (Ichmed) [reply core/src/init.rs:45]: I'll probably switch to a custom Executor anyway so this is just for testing purposes

**Pull request merged**